### DISCUSSION TITLE
Is there master master like replication or similar  strategy available in emqx servers ?

### DISCUSSION THREAD

**Discussion by ajmal-itspe** [emqx/emqx#10565] [General]
Hi!
     I have two emqx servers it is possible to set up the servers in master-master-like replication or a similar strategy? That data generated to one server is automatically generated to other server.

**Comment by qzhuyan** [comment]
If you mean states replication, try use the following config 

```
node {
  db_backend = "mnesia"
  db_role = "core"
}
```

if you mean replicate MQTT message from one to another, checkout bridge mode.

**Comment by SergeTupchiy** [comment]
Hi @ajmal-itspe,

As an example for bridging solution suggested by @qzhuyan above, you can do the following:
- set up two separate EMQX nodes or clusters (referred to as One and Two below)
- create egress MQTT bridge on EMQX One with the following settings:
    - MQTT Broker = EMQX Two
    - egress = true
    - topic = ${topic} 
-  create a rule for the above bridge on EMQX One: `SELECT * FROM "#"`.
